pinot-core/src/main/java/org/apache/pinot/core/util/QueryOptionsUtils.java

```diff
@@ -87,4 +88,12 @@ public static Integer getMinServerGroupTrimSize(Map<String, String> queryOptions
     String minServerGroupTrimSizeString = queryOptions.get(Request.QueryOptionKey.MIN_SERVER_GROUP_TRIM_SIZE);
     return minServerGroupTrimSizeString != null ? Integer.parseInt(minServerGroupTrimSizeString) : null;
   }
+
+  public static boolean isNullHandlingEnabled(Map<String, String> queryOptions) {
+    boolean nullHandlingEnabled = Boolean.parseBoolean(queryOptions.get(Request.QueryOptionKey.ENABLE_NULL_HANDLING));
+    if (nullHandlingEnabled) {
+      Preconditions.checkState(DataTableFactory.getDataTableVersion() >= DataTableFactory.VERSION_4);
```

Comment:
Add some exception message such as: `Null handling cannot be enabled for data table version smaller than 4`